### ISSUE TITLE
Error when trying to use the openshift infrastructure

### ISSUE THREAD

**Issue by abonas** [hawkular/hawkinit#25]
this is the output:
Error when trying to use the openshift infrastructure: Error: Command failed: oc cluster down &> /dev/null ; curl -s https://raw.githubusercontent.com/hawkular/hawkinit/master/lib/configs/openshift/startEphemeral.sh | HAWKULAR_SERVICES_IMAGE=hawkular/hawkular-services:0.36.0.Final TEMPLATE=https://raw.githubusercontent.com/hawkular/hawkinit/master/lib/configs/openshift/hawkular-services-ephemeral.yaml sh

when I ran 'oc cluster down' manually, it works fine.
The oc that I have is v 1.5.0

**Comment by abonas** [comment]
still happening, uninstalled/installed hawkinit. getting the same err

Unable to connect to the server: dial tcp: lookup console.mjelen.pixy.io: no such host
Unable to connect to the server: dial tcp: lookup console.mjelen.pixy.io: no such host
Unable to connect to the server: dial tcp: lookup console.mjelen.pixy.io: no such host
If it failed in the 'Checking container networking' step, try to shut down the cluster, run sudo iptables -F and try again.
Error when trying to use the openshift infrastructure: Error: Command failed: oc cluster down &> /dev/null ; curl -s https://raw.githubusercontent.com/hawkular/hawkinit/master/lib/configs/openshift/startEphemeral.sh | HAWKULAR_SERVICES_IMAGE=hawkular/hawkular-services:latest TEMPLATE=https://raw.githubusercontent.com/hawkular/hawkinit/master/lib/configs/openshift/hawkular-services-ephemeral.yaml sh

@josejulio any idea?

**Comment by josejulio** [comment]
What's your hawkinit version?

I don't really know much about openshift, but it seems like a openshift issue to me, or maybe the templates are incompatible with your openshift version?

@rubenvp8510 do you know what could be the problem?

**Comment by rubenvp8510** [comment]
I don't think this is an issue with templates, it seems to me like something is blocking container network, @abonas  did you tried disabling iptables?

**Comment by abonas** [comment]
@rubenvp8510 yes I did disable them

**Comment by rubenvp8510** [comment]
@abonas  Is there something using ports tcp/8443, udp/53 and udp/8053? If you run `oc cluster up` alone, works?

**Comment by abonas** [comment]
different problems for running oc cluster up

oc cluster up
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... 
   WARNING: Cannot verify Docker version
-- Checking for existing OpenShift container ... OK
-- Checking for openshift/origin:v1.5.1 image ... OK
-- Checking Docker daemon configuration ... FAIL
   Error: did not detect an --insecure-registry argument on the Docker daemon
   Solution:
     Ensure that the Docker daemon is running with the following argument:
     	--insecure-registry 172.30.0.0/16

did not find quickly a way to send the insecure parameter to docker, it doesn't work in command line, probably needs to be added in some file.

**Comment by rubenvp8510** [comment]
@abonas 

add this line

`  "insecure-registries" : ["172.30.0.0/16"]
`
to this file:

`/etc/docker/daemon.json
`

At least this works on my docker, 

```
docker --version
Docker version 1.12.6, build ae7d637/1.12.6
```

**Comment by jkremser** [comment]
wrt `Unable to connect to the server: dial tcp: lookup console.mjelen.pixy.io: no such host` I think, you were doing some tutorial or exercise with openshift and you were remotely connecting with oc to this node, but now it's gone, or the token has expired.

wrt. `insecure-registry`, yes, in case you use the docker CE distribution, the ^ should work. In case, you use the default one that comes with fedora, the file to look into is `/etc/sysconfig/docker` (btw. https://github.com/hawkular/hawkular-services/tree/master/openshift#prerequisites)

To make it even more complicated, they change it on Fedora 26, now the file with the registries is: 
`/etc/containers/registries.conf`

there should be line that looks like: 
```bash
insecure_registries:
  - 172.30.0.0/16
```

Btw. the support for openshift in hawkinit is very basic, it uses the fixed version of the template (not the latest one). Instead of using hawkinit, I'd recommend going into https://github.com/hawkular/hawkular-services/tree/master/openshift and reading the readme. I've created a shell script that should run the ephemeral instance, not sure if it still works though. theute was using it iirc.
